src/cosalib/builds.py

```diff
@@ -65,10 +65,25 @@ def has(self, build_id):
     def is_empty(self):
         return len(self._data['builds']) == 0
 
+    def is_empty_for_arch(self, basearch=None):
+        if not basearch:
+            basearch = get_basearch()
+        for build in self._data['builds']:
+            if basearch in self.get_build_arches(build['id']):
+                return False
+        return True
+
     def get_latest(self):
         # just let throw if there are none
         return self._data['builds'][0]['id']
 
+    def get_latest_for_arch(self, basearch=None):
+        if not basearch:
+            basearch = get_basearch()
+        for build in self._data['builds']:
+            if basearch in self.get_build_arches(build['id']):
+                return build['id']
```

Comment:
Should we have any concern about `self._data['builds']` somehow iterating and not giving us the latest build first in the loop iteration?